### PR TITLE
GTC-1921: Upgrade Postgres Minor Version

### PR DESCRIPTION
**Pull request by gtempus**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

The `dev` aurora cluster is currently configured for `12.4`. AWS no longer supports this minor version.

Issue Number: [GTC-1921](https://gfw.atlassian.net/browse/GTC-1921)

## What is the new behavior?

- The Aurora cluster version in Terraform is the same as the AWS cluster instance (`12.8`)
- Remove the references to `tmaschler`
- Add references to `gtempus`

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: terraform/main.tf [hunk 1]
```diff
@@ -40,6 +40,7 @@ module "postgresql" {
   project                     = var.project_prefix
   rds_backup_retention_period = var.backup_retention_period
   rds_db_name                 = "geostore"
+  rds_version                 = "12.8"
```
Comment (solomon-negusse) [inline terraform/main.tf]: I see you parameterised this at the postgresql module level but hard-coded it at the top level here. If the intention is to enable environment specific version (saw that in commit message unless I misunderstood), we should parameterise it at this level as well.

Comment (gtempus) [reply terraform/main.tf]: Good catch, @solomon-negusse. Yes, I can very well imagine that different versions will be needed for different environments. I'll make that change. 👍

Comment (gtempus) [reply terraform/main.tf]: @solomon-negusse, the change is in commit: 3fd31b36e90e858433ca89646951ef3493506a13

**Review by solomon-negusse — APPROVED** [review]

**Review by dmannarino — APPROVED** [review]
Let's just make sure to do a plan against each environment before merging to make sure TF doesn't want to replace DB.

**Comment by gtempus** [comment]
# Dev Plan
```
 ENV=dev ./scripts/infra plan \-var "gfw_api_token=gfw_api_token" \                       
                        -var "slack_data_updates_hook=slack_hook" \
                        -var "rds_password=password" \
                        -var "rds_password_ro=password" \
                        -var "gfw-gee-export_key=gee_export_key" \
                        -var "planet_api_key=planet_api_key" \
                        -var "jterry_ip=192.168.1.10" \
                        -var "dmannarino_ip=192.168.1.11" \
                        -var "snegusse_ip=192.168.1.12" \
                        -var "gtempus_ip=192.168.1.13" \
                        -var "office_3sc_ip=192.168.1.14" \
                        -var "vpn_3sc_ip=192.168.1.15"
```
result:
```
/usr/local/src/terraform /usr/local/src
Using environment dev and AWS_PROFILE gfw-dev
Starting terraform plan...
Terraform plan - Running pre-plan init
....
  # module.postgresql.aws_rds_cluster.aurora_cluster will be updated in-place
  ~ resource "aws_rds_cluster" "aurora_cluster" {
        apply_immediately                   = true
        arn                                 = "arn:aws:rds:us-east-1:563860007740:cluster:gfw-aurora"
        availability_zones                  = [
            "us-east-1a",
            "us-east-1b",
            "us-east-1d",
        ]
        backtrack_window                    = 0
        backup_retention_period             = 1
        cluster_identifier                  = "gfw-aurora"
        cluster_members                     = [
            "application-autoscaling-ff338b68-5650-4108-9aad-0245a1943ee7",
            "core-aurora-instance-0",
        ]
        cluster_resource_id                 = "cluster-PWRFTOC56PXS45R32FBFEDI2HI"
        copy_tags_to_snapshot               = true
        database_name                       = "geostore"
        db_cluster_parameter_group_name     = "default.aurora-postgresql12"
        db_subnet_group_name                = "main"
        deletion_protection                 = false
      + enable_global_write_forwarding      = false
        enable_http_endpoint                = false
        enabled_cloudwatch_logs_exports     = [
            "postgresql",
        ]
        endpoint                            = "gfw-aurora.cluster-cmwunwkcyxbz.us-east-1.rds.amazonaws.com"
        engine                              = "aurora-postgresql"
        engine_mode                         = "provisioned"
        engine_version                      = "12.8"
        engine_version_actual               = "12.8"
        final_snapshot_identifier           = "core-aurora-cluster"
        hosted_zone_id                      = "Z2R2ITUGPM61AM"
        iam_database_authentication_enabled = false
        iam_roles                           = []
        id                                  = "gfw-aurora"
        kms_key_id                          = "arn:aws:kms:us-east-1:563860007740:key/03cab97d-b476-4d2d-b990-0690f6dc9f47"
      ~ master_password                     = (sensitive value)
        master_username                     = "gfw"
        port                                = 5432
        preferred_backup_window             = "14:00-15:00"
        preferred_maintenance_window        = "sat:16:00-sat:17:00"
        reader_endpoint                     = "gfw-aurora.cluster-ro-cmwunwkcyxbz.us-east-1.rds.amazonaws.com"
        skip_final_snapshot                 = false
        storage_encrypted                   = true
        tags                                = {
            "BuiltBy"     = "Terraform"
            "Environment" = "dev"
            "Job"         = "Geostore"
            "Name"        = "core-Aurora-DB-Cluster"
            "Project"     = "Global Forest Watch"
        }
        tags_all                            = {
            "BuiltBy"     = "Terraform"
            "Environment" = "dev"
            "Job"         = "Geostore"
            "Name"        = "core-Aurora-DB-Cluster"
            "Project"     = "Global Forest Watch"
        }
        vpc_security_group_ids              = [
            "sg-0c47ea95cbc286d60",
        ]

        timeouts {}
    }

  ....
To perform exactly these actions, run the following command to apply:
    terraform apply "terraform.tfplan"

Releasing state lock. This may take a few moments...
Terraform plan finished.
```

**Comment by gtempus** [comment]
# Staging Plan
```
ENV=staging ./scripts/infra plan \-var "gfw_api_token=gfw_api_token" \
                        -var "slack_data_updates_hook=slack_hook" \
                        -var "rds_password=password" \
                        -var "rds_password_ro=password" \
                        -var "gfw-gee-export_key=gee_export_key" \
                        -var "planet_api_key=planet_api_key" \
                        -var "jterry_ip=192.168.1.10" \
                        -var "dmannarino_ip=192.168.1.11" \
                        -var "snegusse_ip=192.168.1.12" \
                        -var "gtempus_ip=192.168.1.13" \
                        -var "office_3sc_ip=192.168.1.14" \
                        -var "vpn_3sc_ip=192.168.1.15"
```
result:
```
/usr/local/src/terraform /usr/local/src
Using environment staging and AWS_PROFILE gfw-staging
Starting terraform plan...
....

  # module.postgresql.aws_rds_cluster.aurora_cluster will be updated in-place
  ~ resource "aws_rds_cluster" "aurora_cluster" {
        apply_immediately                   = true
        arn                                 = "arn:aws:rds:us-east-1:274931322839:cluster:gfw-aurora"
        availability_zones                  = [
            "us-east-1a",
            "us-east-1b",
            "us-east-1d",
        ]
        backtrack_window                    = 0
        backup_retention_period             = 1
        cluster_identifier                  = "gfw-aurora"
        cluster_members                     = [
            "application-autoscaling-698c9c01-db99-4430-a97a-6baaae853dc6",
            "core-aurora-instance-0",
        ]
        cluster_resource_id                 = "cluster-FX4UXHDFY2KKJC25ZW3RB6WHAU"
        copy_tags_to_snapshot               = true
        database_name                       = "geostore"
        db_cluster_parameter_group_name     = "default.aurora-postgresql12"
        db_subnet_group_name                = "main"
        deletion_protection                 = false
        enable_global_write_forwarding      = false
        enable_http_endpoint                = false
        enabled_cloudwatch_logs_exports     = [
            "postgresql",
        ]
        endpoint                            = "gfw-aurora.cluster-cljrqsduwhdo.us-east-1.rds.amazonaws.com"
        engine                              = "aurora-postgresql"
        engine_mode                         = "provisioned"
        engine_version                      = "12.8"
        engine_version_actual               = "12.8"
        final_snapshot_identifier           = "core-aurora-cluster"
        hosted_zone_id                      = "Z2R2ITUGPM61AM"
        iam_database_authentication_enabled = false
        iam_roles                           = []
        id                                  = "gfw-aurora"
        kms_key_id                          = "arn:aws:kms:us-east-1:274931322839:key/8c47651e-a11f-444d-8013-9334fbac400f"
      ~ master_password                     = (sensitive value)
        master_username                     = "gfw"
        port                                = 5432
        preferred_backup_window             = "14:00-15:00"
        preferred_maintenance_window        = "sat:16:00-sat:17:00"
        reader_endpoint                     = "gfw-aurora.cluster-ro-cljrqsduwhdo.us-east-1.rds.amazonaws.com"
        skip_final_snapshot                 = false
        storage_encrypted                   = true
        tags                                = {
            "BuiltBy"     = "Terraform"
            "Environment" = "staging"
            "Job"         = "Geostore"
            "Name"        = "core-Aurora-DB-Cluster"
            "Project"     = "Global Forest Watch"
        }
        tags_all                            = {
            "BuiltBy"     = "Terraform"
            "Environment" = "staging"
            "Job"         = "Geostore"
            "Name"        = "core-Aurora-DB-Cluster"
            "Project"     = "Global Forest Watch"
        }
        vpc_security_group_ids              = [
            "sg-0eac0da129448b72e",
        ]
    }

  ....
Releasing state lock. This may take a few moments...
Terraform plan finished.
```

**Comment by gtempus** [comment]
I noticed this rds cluster change for `DEV` @dmannarino. Is that expected?
Screen Shot 2022-07-19 at 3 06 42 PM

**Pull request merged**